returnn/training.py

```diff
@@ -788,3 +792,63 @@ def run(self):
                 self._out_model_dir.get_path(), "checkpoint.data-00000-of-00001"
             ),
         )
+
+
+class AverageTFCheckpointsJob(Job):
+    """
+    Compute the average of multiple specified Tensorflow checkpoints using the tf_avg_checkpoints script from Returnn
+    """
+
+    def __init__(
+        self,
+        model_dir: tk.Path,
+        epochs: List[Union[int, tk.Variable]],
+        returnn_python_exe: tk.Path,
+        returnn_root: tk.Path,
+    ):
+        """
+
+        :param model_dir: model dir from `ReturnnTrainingJob`
+        :param epochs: manually specified epochs or `out_epoch` from `GetBestEpochJob`
+        :param returnn_python_exe: file path to the executable for running returnn (python binary or .sh)
+        :param returnn_root: file path to the RETURNN repository root folder
+        """
+        self.model_dir = model_dir
+        self.epochs = epochs
+        self.returnn_python_exe = returnn_python_exe
+        self.returnn_root = returnn_root
+
+        self._out_model_dir = self.output_path("model", directory=True)
+        self.out_checkpoint = Checkpoint(self.output_path("model/average.index"))
+
+        self.rqmt = {"cpu": 1, "time": 0.5, "mem": max(2 * len(epochs), 4)}
```

Comment:
isn't `2*len(epochs)` always larger or equal to `4` -- assuming that you provide at least 2 epochs. 
I would rather specify a maximum memory requirement (if that is possible). e.g.
```suggestion
        self.rqmt = {"cpu": 1, "time": 0.5, "mem": min(2 * len(epochs), 16)}
```

Or is the mem rqmt strictly linear at 2GB per epoch s.t. we cannot merge arbitrary many epochs together?

Reply:
But you are right that I can drop the max, as no one will ever run with less than 2.

Reply:
I see.. then maybe the implementation of the script has to be altered if memory turns out to be a problem in the future.

There is still the check_engine_limits function to reduce the mem request.